test/extended/cpu_partitioning/crio.go

```diff
@@ -0,0 +1,314 @@
+			// that contain workload annotations. In the event that they do, we allow the logic to continue to validate
+			// CPUSets.
+			if isClusterCPUPartitioned {
+				o.Expect(crioData).ToNot(o.BeEmpty(), "error getting crio container data, container data should not be empty")
+			}
+
+			// CPU Core counts are numbers 0-n, we minus one since Status.Capacity.Cpu count is numbered conventionally.
+			// We generate a wide open CPUSet to use for comparison.
+			totalCoreCount := node.Status.Capacity.Cpu().Value() - 1
+			parsedFullNodeCPUSet, err := cpuset.Parse(fmt.Sprintf("0-%d", totalCoreCount))
+			o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error creating full node cpuset, check totalCoreCount of (%d)", totalCoreCount))
+
+			expectedCPUSet := ""
+
+			g.By(fmt.Sprintf("fetching crio data from node (%s)", node.Name))
+			if _, isMaster := node.Labels[nodeMasterLabel]; isMaster {
+				expectedCPUSet = nodeRoleCPUSet[nodeMasterLabel]
+			} else if _, isWorker := node.Labels[nodeWorkerLabel]; isWorker {
+				expectedCPUSet = nodeRoleCPUSet[nodeWorkerLabel]
+			}
+
+			g.By(fmt.Sprintf("checking for expected CPUSet %s", expectedCPUSet))
+
+			for _, containerInfo := range crioData {
+				o.Expect(containerInfo.ContainerCPUSet).To(
+					o.Equal(expectedCPUSet),
+					"cpusets do not match between container and desired")
+
+				parsedContainer, err := cpuset.Parse(containerInfo.ContainerCPUSet)
+				o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error parsing crio container cpuset %s", containerInfo.ContainerCPUSet))
+				parsedHost, err := cpuset.Parse(containerInfo.HostCPUSet)
+				o.Expect(err).ToNot(o.HaveOccurred(), fmt.Sprintf("error parsing host cpuset %s", containerInfo.HostCPUSet))
+
+				// If the expected CPUSet is not empty, we make sure the host is respecting it.
+				// Otherwise, empty CPUSets mean wide open, so we check the processes are not being limited.
+				if expectedCPUSet != "" {
+					o.Expect(parsedContainer.Equals(parsedHost)).To(o.BeTrue(), fmt.Sprintf("expected: %s got: %s", parsedContainer, parsedHost))
+				} else {
+					o.Expect(parsedHost.Equals(parsedFullNodeCPUSet)).To(o.BeTrue(), fmt.Sprintf("expected host CPUSet: %s got: %s", parsedHost, parsedFullNodeCPUSet))
+				}
+
+				// If we are in a CPU Partitioned cluster, containers MUST be annotated with the correct CPU Share at the CRIO level
+				// and the desired annotation cpu shares must equal the crio config cpu shares
+				if isClusterCPUPartitioned {
+					share, err := containerInfo.getAnnotationCPUShare()
+					o.Expect(err).ToNot(o.HaveOccurred(), "failed to get cpushares annotation json", err)
+					o.Expect(share).To(o.Equal(containerInfo.ContainerCPUShare), "cpushares do not match between crio config and desired")
+				}
+			}
+		}
+	})
+})
+
+func getExpectedCPUSetConstraints(ctx context.Context, oc *exutil.CLI, isClusterPartitioned bool) (map[string]string, error) {
```

Comment:
minor: nonblocking: I'd return a plain object instead of packing return values in a map.

Reply:
Good idea, updated to use a struct that captures the CPUSets for masters and workers.